src/map/unit.cpp

```diff
@@ -2965,15 +2965,15 @@ static int32 unit_attack_timer_sub(struct block_list* src, int32 tid, t_tick tic
 	}
 
 	if( !battle_check_range(src,target,range) ) {
-	  	// Within range, but no direct line of attack
-		if( ud->state.attack_continue ) {
-			if(ud->chaserange > 2)
-				ud->chaserange-=2;
-
+		// Within range, but no direct line of attack
+		// This code can usually only be reached if OFFICIAL_WALKPATH is disabled
+		if (ud->state.attack_continue && ud->chaserange > 1) {
+			ud->chaserange = std::max(1, ud->chaserange - 2);
 			unit_walktobl(src,target,ud->chaserange,ud->state.walk_easy|2);
+			return 1;
```

Comment:
Based on your explanations in discord:
Should we not return the result of unit_walktobl here?
Because if you fail to start walking, then you actually should give up or not?
Just like you changed it below.